lib/cell.rb

```diff
@@ -0,0 +1,25 @@
+class Cell
+  attr_reader :coordinate, :ship
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @ship = nil
+    @hit = false
+  end
+
+  def empty?
+    @ship == nil
+  end
+
+  def fired_upon?
+    @hit
+  end
+
+  def fire_upon
+    @hit = true
+    @ship.hit
+  end
+
+  def place_ship(ship)
```

Comment:
Light suggestion here as well. To change the method parameter from ship to ship_name or ship_type just to make a clear distinction between that variable and our ship method.

Reply:
I can see how that is confusing will make some changes.

Reply:
Now that I think about it the parameter represents a ship object so naming it ship_name or ship_type would be misleading. Is ship_object ok for now? Naming is hard.

Reply:
Naming is hard indeed. ship_object is good.